applications/api-cat/src/main/java/no/acat/restapi/ParseApiSpecTilApiDocumentController.java

```diff
@@ -0,0 +1,23 @@
+package no.acat.restapi;
+
+
+import lombok.AllArgsConstructor;
+import no.acat.model.ApiDocument;
+import no.acat.model.ApiSource;
+import no.acat.spec.converters.ParseApiSpecToApiDocument;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/apiparser")
+@AllArgsConstructor
+public class ParseApiSpecTilApiDocumentController {
+
+    private ParseApiSpecToApiDocument parseApiSpecToApiDocument;
+
+    @PostMapping( produces = "application/json")
+    public ApiDocument parseApiSpec(@RequestBody ApiSource apiSource) {
```

Comment:
Jeg liker virkelig ikke ApiSource og bruk av den. Hva med:


    @PostMapping( produces = "application/json")
    public ApiDocument parseApiSpec(@RequestBody Map<String,String> apiSource) {
        if (apiSource.containsKey("url")) {
            return parseApiSpecToApiDocument.parseApiSpecFromUrl(apiSource.get("url"));
        } else if (apiSource.containsKey("data")) {
            return parseApiSpecToApiDocument.parseApiSpecFromData(apiSource.get("data"));
        } else {
            return null; //throw Exception, whatever
        }
    }

Reply:
Jeg er bite litt uenig her - den typ kan være brukelig i swagger beskrivelse. vi må ha en mote å dokumentere body parameters. Je kan tenke om to metoder - en er å bruke class til requestBody, en ant møte er å bruke annotasjoner. Det ser litt lettere ut med class, men uansett, de må testes med swagger.